readers-frontend/src/components/SearchInterface.astro

```diff
@@ -12,37 +14,106 @@ import { Icon } from "astro-icon";
   >
     <Icon name="ic:baseline-close" />
   </button>
-  <div class="mx-auto my-auto p-4">
-    <form id="searchForm" class="flex flex-row items-center">
+  <div class="mx-auto my-auto flex w-full flex-col p-4">
+    <form
+      id="searchForm"
+      class="mx-auto mb-10 mt-10 flex flex-row items-center md:mt-0"
+    >
       <input
         id="searchBox"
         type="text"
-        class="w-min border-b-2 border-black bg-whiteish-100 outline-none sm:text-3xl md:text-4xl lg:text-5xl xl:text-7xl"
+        class="bg-whiteish-100 w-min border-b-2 border-black outline-none sm:text-3xl md:text-4xl lg:text-5xl xl:text-7xl"
         placeholder="What are you looking for?"
         size="25"
       />
-      <button class="w-10 md:w-14 lg:w-20">
-        <Icon name="ic:sharp-search" class="text-black" />
+      <button class="relative h-10 w-10 md:h-14 md:w-14 lg:h-20 lg:w-20">
+        <Icon name="ic:sharp-search" class="h-full w-full text-black" />
       </button>
     </form>
   </div>
 </div>
 
 <script>
+  // Show / hide
   const closeBtn = document.getElementById("searchCloseBtn") as HTMLElement;
-  const form = document.getElementById("searchForm") as HTMLFormElement;
   const searchInterface = document.getElementById(
     "searchInterface"
   ) as HTMLElement;
 
   closeBtn.addEventListener("click", () => {
-    console.log("click");
     searchInterface.classList.add("invisible", "opacity-0");
   });
+
+  // Form
+  const form = document.getElementById("searchForm") as HTMLFormElement;
+  const input = document.getElementById("searchBox") as HTMLInputElement;
+
   form.addEventListener("submit", (ev) => {
     ev.preventDefault();
-    const input = document.getElementById("searchBox") as HTMLInputElement;
 
-    // TODO: perform api request with input.value
+    const query = encodeURIComponent(input.value);
+    if (query.length > 0) {
+      window.location.replace(
+        `${import.meta.env.BASE_URL}search?q=${query}`
+      );
+    }
   });
 </script>
+
+<style is:global={true}>
+  .category-text {
+    font-family: "arno-pro-display", serif;
+    font-weight: 700;
+    font-style: normal;
+  }
+  /* Article category text colour */
```

Comment:
Can you change these to be by category name? It just makes it easier to update them. Also, what do you think about moving them to the layout component?

Reply:
I've made it so the classes are computed at runtime. I don't think we should keep it like that but for now I think it works. Ideally I'd like to have a CSS file generate at build time with the classes

Reply:
Or maybe just pass the colour from the API at this point?

Reply:
I see what you did. yeah ideally we'd do it at build time. I did some digging and currently that doesn't seem to be supported by Astro. I say let's add it to the trello board as a ticket and we can come back to it closer to launch